ml/amaranth_lib.py

```diff
@@ -0,0 +1,179 @@
+# Lint as: python3
+"""This module defines helper functions for use in ml/main.py.
+
+This module is in charge of encapsulating commonly used methods or well-defined
+functions away from ml/main.py for simplicity and readability.
+"""
+
+from typing import Iterable, Sized
+import pandas as pd
+from tensorflow import keras
+
+
+def combine_dataframes(index: str, *dataframes: pd.DataFrame):
+  """Takes a DataFrame index, and a number of DataFrames to join based on that index.
+
+  Args:
+    index (str): The title of the column to combine the DataFrames on
+    *dataframes (pd.DataFrame): Dataframes to combine
+
+  Returns:
+    combined_dataframe (pd.DataFrame): The final combined dataframe with it's
+    index set to 'index'
+  Raises:
+    KeyError: If any dataframes don't have the 'index' column
+  """
+
+  if not dataframes:
+    return pd.DataFrame()
+
+  combined_dataframe = dataframes[0].set_index(index)
+  for df in dataframes[1:]:
+    combined_dataframe = combined_dataframe.join(df.set_index(index))
+
+  return combined_dataframe
+
+
+def get_calorie_data(dataframe: pd.DataFrame, units: str):
+  """Returns a copy of a DataFrame with only calorie data in the specified units.
+
+  Returns a copy of dataframe where the 'name' column must be equal to 'Energy'
+  and the 'unit_name' column must be equal to 'units' (ignoring case). As such,
+  both of these columns must be present, otherwise you will get a KeyError.
+
+  Args:
+    dataframe (pd.DataFrame): The DataFrame to extract calorie data from
+    units (str): The desired units of the calorie data
+
+  Returns:
+    calorie_data (pd.DataFrame): All calorie-related data present in
+    'dataframe' with 'units' units.
+
+  Raises:
+    KeyError: If either 'name' or 'unit_name' columns are absent in 'dataframe'
+  """
+
+  calorie_data = dataframe[
+      (dataframe['name'] == 'Energy')
```

Comment:
might want to separate this term out, since it's a bit of "magic"

Reply:
Good idea! Consider the magic ✨ gone ✨ https://github.com/googleinterns/amaranth/pull/7/commits/4b604ce672f013058fd9acd3ec5f926df3bacafa